angular-workspace/projects/ni/nimble-angular/table-column/number-text/tests/nimble-table-column-number-text.directive.spec.ts

```diff
@@ -0,0 +1,679 @@
+import { Component, ElementRef, ViewChild } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NimbleTableModule } from '../../../table/nimble-table.module';
+import { TableColumnSortDirection } from '../../nimble-table-column-base.directive';
+import { NimbleTableColumnNumberTextDirective, NumberTextFormat, TableColumnNumberText } from '../nimble-table-column-number-text.directive';
+import { NimbleTableColumnNumberTextModule } from '../nimble-table-column-number-text.module';
+
+describe('NimbleTableColumnNumberText', () => {
+    describe('module', () => {
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                imports: [NimbleTableColumnNumberTextModule]
+            });
+        });
+
+        it('custom element is defined', () => {
+            expect(customElements.get('nimble-table-column-number-text')).not.toBeUndefined();
+        });
+    });
+
+    describe('with default configuration', () => {
+        @Component({
+            template: `
+                <nimble-table>
+                    <nimble-table-column-number-text #column field-name="field1">
+                    </nimble-table-column-number-text>
+                </nimble-table>
+            `
+        })
+        class TestHostComponent {
+            @ViewChild('column', { read: NimbleTableColumnNumberTextDirective }) public directive: NimbleTableColumnNumberTextDirective;
+            @ViewChild('column', { read: ElementRef }) public elementRef: ElementRef<TableColumnNumberText>;
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleTableColumnNumberTextDirective;
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleTableColumnNumberTextModule, NimbleTableModule]
+            });
+
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+        });
+
+        it('has valid configuration by default', () => {
+            expect(directive.checkValidity()).toBeTrue();
+            expect(directive.validity.invalidDecimalDigits).toBeFalse();
+        });
+
+        it('has invalid configuration with format set to decimal with invalid decimal-digits', () => {
+            directive.format = NumberTextFormat.decimal;
+            directive.decimalDigits = 100;
+            expect(directive.checkValidity()).toBeFalse();
+            expect(directive.validity.invalidDecimalDigits).toBeTrue();
+        });
```

Comment:
We could, but I'm not sure how important that is. We have tests like that for the web component, so I'm not convinced we need to duplicate all those tests here. The main goal of the Angular tests is to make sure using the component in Angular correctly applies & retrieves the state from the underlying web component.

@m-akinc, how strongly do you feel about this? If you don't feel strongly, then I'm inclined to not add an additional test because I think as-is has consistent test coverage with our other Angular directives.